src/JustSaying/AwsTools/MessageHandling/SnsMessagePublisher.cs

```diff
@@ -150,4 +157,109 @@ public virtual InterrogationResult Interrogate()
+
+                        foreach (var message in response.Successful)
+                        {
+                            _logger.LogInformation(
+                                "Published message {MessageId} of type {MessageType} to {DestinationType} '{MessageDestination}'.",
+                                message.Id,
+                                message.GetType().FullName,
+                                "Topic",
+                                request.TopicArn);
+                        }
+                    }
+
+                    if (response.Failed.Count > 0 && _logger.IsEnabled(LogLevel.Error))
+                    {
+                        _logger.LogError(
+                            "Fail to published batch of {MessageCount} to {DestinationType} '{MessageDestination}'.",
+                            response.Failed.Count,
+                            "Topic",
+                            request.TopicArn);
+
+                        foreach (var message in response.Failed)
+                        {
+                            _logger.LogError(
+                                "Fail to published message {MessageId} to {DestinationType} '{MessageDestination}' with error code: {ErrorCode} is error on BatchAPI: {IsBatchAPIError}.",
+                                message.Id,
+                                "Topic",
+                                request.TopicArn,
+                                message.Code,
+                                message.SenderFault);
+                        }
+                    }
+                }
+            }
+
+            if (MessageBatchResponseLogger != null)
+            {
+                var responseData = new MessageBatchResponse
+                {
+                    SuccessfulMessageIds = response?.Successful.Select(x => x.MessageId).ToArray(),
+                    FailedMessageIds = response?.Failed.Select(x => x.Id).ToArray(),
+                    ResponseMetadata = response?.ResponseMetadata,
+                    HttpStatusCode = response?.HttpStatusCode,
+                };
+
+                MessageBatchResponseLogger(responseData, chunk);
+            }
+        }
+    }
+
+    private bool ClientExceptionHandler(Exception ex, IReadOnlyCollection<Message> messages) => _handleBatchException?.Invoke(ex, messages) ?? false;
+
+    private PublishBatchRequest BuildPublishBatchRequest(Message[] message, PublishMetadata metadata)
+    {
+        return new PublishBatchRequest
+        {
+            TopicArn = Arn,
+            PublishBatchRequestEntries = message.Select(message => new PublishBatchRequestEntry
+                {
+                    Id = message.UniqueKey(),
+                    Subject = _messageSubjectProvider.GetSubjectForType(message.GetType()),
+                    Message = _serializationRegister.Serialize(message, serializeForSnsPublishing: true),
+                    MessageAttributes = BuildMessageAttributes(metadata)
+                })
```

Comment:
This would be better to do manually in a loop as there's a lot of methods being called here, which makes it more difficult to debug if it's a lambda inside a `Select()`.